avalanche/training/templates/supervised.py

```diff
@@ -241,23 +247,50 @@ def make_eval_dataloader(
         if parse_version(torch.__version__) >= parse_version("1.7.0"):
             other_dataloader_args["persistent_workers"] = persistent_workers
 
+        sampler = None
+        if DistributedHelper.is_distributed:
+            sampler = DistributedSampler(
+                self.adapted_dataset, shuffle=False, drop_last=False)
+
         self.dataloader = DataLoader(
             self.adapted_dataset,
             num_workers=num_workers,
             batch_size=self.eval_mb_size,
             pin_memory=pin_memory,
+            sampler=sampler,
+            shuffle=False,
+            drop_last=False,
             **other_dataloader_args
         )
 
+    @final
     def forward(self):
-        """Compute the model's output given the current mini-batch."""
+        """
+        Compute the model's output given the current mini-batch.
+
+        This method should not be overridden by child classes.
+        Consider overriding :meth:`_forward` instead.
+        """
+        with self.use_local_input_batch():
+            return self._forward()
+
+    def _forward(self):
+        """Implementation of the forward pass."""
         return avalanche_forward(self.model, self.mb_x, self.mb_task_id)
 
+    @final
     def model_adaptation(self, model=None):
```

Comment:
The reason here is that we want the users to override `_model_adaptation` instead of `model_adaptation`. The _model_adaptation is then called from within the `with `self.use_local_model()` context. This is needed to ensure the users are changing the local model instead of the `model` (which may be the wrapped one).